src/session/session_api.c

```diff
@@ -166,8 +166,7 @@ __session_alter(WT_SESSION *wt_session, const char *uri, const char *config)
 	WT_WITH_CHECKPOINT_LOCK(session,
 	    WT_WITH_SCHEMA_LOCK(session,
 		ret = __wt_schema_worker(session, uri, __wt_alter, NULL, cfg,
-		WT_BTREE_ALTER |
-		WT_DHANDLE_DISCARD_FORCE | WT_DHANDLE_EXCLUSIVE)));
```

Comment:
I should add that there's one remaining use of `WT_DHANDLE_DISCARD_FORCE`, which is in LSM.  I'm inclined to try to eliminate it and find some cleaner way for LSM to get handles purged from cache.